org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JSONUtility.java

```diff
@@ -55,6 +56,9 @@ private static <T> T toModel(Gson gson, Object object, Class<T> clazz) {
 		if (object instanceof JsonElement json) {
 			return gson.fromJson(json, clazz);
 		}
+		if (object instanceof LinkedTreeMap) {
+			return gson.fromJson(gson.toJson(object), clazz);
+		}
```

Comment:
`LinkedTreeMap` is internal to gson so we should avoid using it, mainly because it generates some warnings. However, you don't even need to use this if the client converts the object to a JSON string prior to sending in the delegate handler. In that case it'll be handled by the `object instanceof String json` from below.

I'll make the change.